bittide/src/Bittide/ScatterGather.hs

```diff
@@ -31,7 +47,7 @@ scatterUnit ::
   , KnownNat memDepth, 1 <= memDepth
   , KnownNat frameWidth) =>
   -- | Configuration for the 'calendar'.
-  CalendarConfig nBytes addrW (CalendarEntry memDepth) ->
+  CalendarConfig nBytes addrW (Index memDepth) ->
```

Comment:
Shouldn't this take a `ScatterConfig`?

Reply:
The `GADT`s are used to hide implementations details where they are not necessary and be able to differentiate  between the S/G unit calendars.
The differentiation would be nice to have, but `memDepth` is present in the API of the `scatterUnit` and gatherUnit`. I could change it, but I'd have to add a run-time check that verifies whether the tyVars are the same.

Reply:
But you're right, not having access to the `memDepth` inside the `ScatterConfig` would be problematic here